bot/cogs/tags.py

```diff
@@ -0,0 +1,266 @@
+                tag_name
+                and tag_name in self.tag_cooldowns
+                and (now - self.tag_cooldowns[tag_name]["time"]) < TAG_COOLDOWN
+                and self.tag_cooldowns[tag_name]["channel"] == ctx.channel.id
+            )
+
+            if cooldown_conditions:
+                return True
+            return False
+
+        if _command_on_cooldown(tag_name):
+            time_left = TAG_COOLDOWN - (time.time() - self.tag_cooldowns[tag_name]["time"])
+            print(f"That command is currently on cooldown. Try again in {time_left:.1f} seconds")
+            return
+
+        embed = Embed()
+        tags = []
+
+        tag_data = await self.get_tag_data(tag_name)
+
+        # If we found something, prepare that data
+        if tag_data:
+            embed.colour = Colour.blurple()
+
+            if tag_name:
+                embed.title = tag_name
+                self.tag_cooldowns[tag_name] = {
+                    "time": time.time(),
+                    "channel": ctx.channel.id
+                }
+
+            else:
+                embed.title = "**Current tags**"
+
+            if isinstance(tag_data, list):
+                tags = [f"**»**   {tag['tag_name']}" for tag in tag_data]
+                tags = sorted(tags)
+
+            else:
+                embed.description = tag_data['tag_content']
+
+        # If not, prepare an error message.
+        else:
+            embed.colour = Colour.red()
+            embed.title = "There are no tags in the database!"
+
+            if isinstance(tag_data, dict):
+                embed.description = f"Unknown tag: **{tag_name}**"
+
+            if tag_name:
+                embed.set_footer(text="To show a list of all tags, use bot.tags.get()")
+                embed.title = "Tag not found!"
+
+        # Paginate if this is a list of all tags
+        if tags:
+            return await paginate(
+                (lines for lines in tags),
+                ctx, embed,
+                footer_text="To show a tag, type bot.tags.get <tagname>",
+                empty=False,
+                max_size=200
+            )
+
+        return await ctx.send(embed=embed)
+
+    @with_role(ADMIN_ROLE, OWNER_ROLE, MODERATOR_ROLE)
+    @command(name="tags.set()", aliases=["tags.set", "tags.add", "tags.add()", "tags.edit", "tags.edit()"])
+    async def set(self, ctx: Context, tag_name: str, tag_content: str):
+        """
+        Create a new tag or edit an existing one.
+
+        :param ctx: discord message context
+        :param tag_name: The name of the tag to create or edit.
+        :param tag_content: The content of the tag.
+        """
+
+        embed = Embed()
+
+        if "\n" in tag_name:
+            embed.colour = Colour.red()
+            embed.title = "Please don't do that"
+            embed.description = "Don't be ridiculous. Newlines are obviously not allowed in the tag name."
+
+        else:
+            if tag_name and tag_content:
```

Comment:
This could be reorganised, for example since the `else` here contains a return, you could invert the conditional and drop the extra branch.

```py
if not (tag_name and tag_content):
    # ....
    return

tag_data = ...
```